frontend/src/components/Participants.vue

```diff
@@ -0,0 +1,286 @@
+  },
+  validations: {
+    form: {
+      email: {
+        required,
+        email,
+        emailExists(value) {
+          return !Object.values(this.allParticipants).find(
+            user => user.email === value
+          );
+        }
+      },
+      github: {
+        required,
+        githubExists(value) {
+          return !Object.values(this.allParticipants).find(
+            user => user.github === value
+          );
+        }
+      },
+      first_name: {
+        required,
+        minLength: minLength(3),
+        maxLength: maxLength(25)
+      },
+      last_name: {
+        required,
+        minLength: minLength(3),
+        maxLength: maxLength(25)
+      },
+      slack: {
+        minLength: minLength(3),
+        maxLength: maxLength(25)
+      },
+      phone: {
+        numeric,
+        minLength: minLength(9),
+        maxLength: maxLength(9)
+      }
+    }
+  },
+  methods: {
+    onSubmit() {
+      if (!this.$v.form.$invalid) {
+        if (this.form.phone === '') this.form.phone = null;
+        if (this.form.slack === '') this.form.slack = null;
+        const newParticipantData = this.form;
+
+        this.$store
+          .dispatch('participant/createParticipant', newParticipantData)
+          .then(status => {
+            if (status === 201) {
+              this.successAlert = true;
+              this.$refs.form.reset();
+              this.$v.form.$reset();
+              setTimeout(() => (this.successAlert = false), 5000);
+            }
+          });
+      }
+    },
+    validateSlack($event) {
+      this.$v.form.slack.$touch();
+      if (
+        Object.values(this.allParticipants).find(
+          user => user.slack === $event.target.value
+        )
+      )
+        this.warningAlert = true;
+      this.warningMessage =
+        'This slack nick already exists. Please suggest change of the nick to avoid confusion on Slack';
+      setTimeout(() => (this.warningAlert = false), 10000);
+    },
+    validateLastName($event) {
+      this.$v.form.slack.$touch();
+      if (
+        Object.values(this.allParticipants).find(
+          user => user.last_name === $event.target.value
+        )
+      )
+        this.warningAlert = true;
+      this.warningMessage =
+        'This last name already exists. Please verify if a person is actually a new paricipant';
+      setTimeout(() => (this.warningAlert = false), 10000);
+    },
+    validatePhone($event) {
+      this.$v.form.phone.$touch();
+      if (
+        Object.values(this.allParticipants).find(
+          user => user.phone === $event.target.value
+        )
+      )
+        this.warningAlert = true;
+      this.warningMessage =
+        'This phone number already exists. Please verify if a person is actually a new paricipant';
+      setTimeout(() => (this.warningAlert = false), 10000);
+    }
+  },
+  computed: {
+    ...mapGetters('participant', ['getError']),
+    ...mapState('participant', {
```

Comment:
That solution is working, but as far as I know better practice is to access state with getters, not directly.
This should work e.g.
`...mapGetters('participant', {getError: 'getError', allParticipants: 'getParticipants'})`

Reply:
Done